### PR TITLE
Honor WSK_CONFIG_FILE if variable is set

### LINKED ISSUE #809: wskdeploy does not respect WSK_CONFIG_FILE
if you set `WSK_CONFIG_FILE`, the `wsk` cli will use its settings for wskprops. It appears that wskdeploy ignores this setting and always uses `~/.wskprops`.

### PR DESCRIPTION
**Pull request by alvarolopez**
The OpenWhisk CLI (wsk) allows to set the WSK_CONFIG_FILE to point to
the configuration file. wskdeploy does not honor this environment
variable and it only relies on the --config flag. This change allows to
use the environment variable (if present), making it consistent with the
wsk CLI behaviour.

Fixes #809
Fixes #436

### CODE REVIEW

**Comment by alvarolopez** [comment]
Any chance that this can get some attention?

**Comment by mrutkows** [comment]
@alvarolopez Hi Alvaro, if you can get the Travis to work, I am prepared to review it.

**Comment by mrutkows** [comment]
@alvarolopez the Travis build failure is now here: https://travis-ci.org/apache/openwhisk-wskdeploy/builds/548118086?utm_source=github_status&utm_medium=notification

I restarted the build to assure logs are current.

File: cmd/root.go [hunk 1]
```diff
@@ -102,21 +102,32 @@ func init() {
...
+    // 1. --config 
+    // 2. ENV variable WSK_CONFIG_FILE
+    // 3. Default $HOME/.wskprops
+	cfgFiles := []string{
+		utils.Flags.CfgFile,
+		os.Getenv("WSK_CONFIG_FILE"),
+        defaultPath,
+	 }
+
+	for _, cfgFile := range cfgFiles {
```
Comment (mrutkows) [inline cmd/root.go]: If indeed there are 3 discrete places we look for a valid config. file (and each should have a single entry/key and associated value), then using an array to loop upon with a break stmt. seems overkill.

Comment (mrutkows) [reply cmd/root.go]: validating each in order and discontinuing search (short circuit) on first success seems much more efficient.

Comment (alvarolopez) [reply cmd/root.go]: Well, I am not a gopher, therefore I do not know if there is any better method to perform it. What would you suggest to do?

**Comment by mrutkows** [comment]
@alvarolopez The build is failing as the linter does not like your go formatting (indentation); could you please fix that so we can review and work towards merge?

File: cmd/root.go (hunk 1 above)
Comment (mrutkows) [reply cmd/root.go]: Can you please clarify "not a gopher" for me?

Comment (alvarolopez) [reply cmd/root.go]: I mean that I an not familiar with Go, therefore I do not know if there any more efficient ways of reading the configuration files than trying to read them in order and breaking the loop on first success.

**Comment by alvarolopez** [comment]
> @alvarolopez The build is failing as the linter does not like your go formatting (indentation); could you please fix that so we can review and work towards merge?

I have cleaned the code (mixed spaces and tabs, my bad) and the linter does not complain. However, the build is still failing, but this is a master branch failure, not introduced by this code.

**Comment by mrutkows** [comment]
@alvarolopez it appears that our integration tests are failing due to a change in the supported runtimes list (not under our control).  I will attempt to provide a PR to fix after contemplating the best future-proof approach.

**Comment by mrutkows** [comment]
The runtime fix in this PR: https://github.com/apache/openwhisk-wskdeploy/pull/1070 should unblock this one.

**Pull request merged**